lib/PipelineC/PipelineC.cpp

```diff
@@ -69,6 +69,8 @@ class ExistingOrNew {
 
 static void llvmErrorToRpError(llvm::Error Error, rp_error *Out) {
   if (Out == nullptr) {
+    // TODO: this is suspicions, isn't it an indication that callee is
+    // doing something nasty? Is it ok to just ignore it?
```

Comment:
That's part of the `PipelineC` calling conventions, if you see the callers of this functions (e.g. `_rp_manager_run_analysis`), they pass through the `rp_error *` variable. The callee of these can opt-in to receive errors by passing the `rp_error *`, whereas they get ignored if `NULL` is provided as an argument. (Essentially the `rp_error *` constitutes an optional return value, but due to C structure that was one of the saner ways to implement it)

Reply:
Done:
```
  // In the context of `lib/Pipeline`, both context and containers are default
  // constructible. As such, we just skip loading files that do not exist,
  // ending up with the default state.
```